### PR TITLE
Refactor/deanonymize-get-prop-funcs

### PR DESCRIPTION
**Pull request by mrufsvold**
I don't want to break the api, so I created a simple get_property function that looks up the correct function and then returns an anonymous function with "this" inserted in the first position. 

I commented out a warning log about deprecation because basically every function call in the code base hits that warning at the moment. But once everything is moved to using direct function calls, we can turn on the warning to track down the last of them.

To be clear: I didn't change any logic here. Just renamed functions to snake_case and moved them out of `get_property`

### CODE REVIEW

File: src/Camera.jl [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 mutable struct Camera
     dimensions::Vector2
     offset::Vector2f
     position::Vector2f
     startingCoordinates::Vector2f
-    target#EntityModule.Entity
+
+    target::Union{
+        Ptr{Nothing}, 
+        # JulGame.EntityModule.Entity,
```
Comment (mrufsvold) [inline src/Camera.jl:13]: I originally assumed the target of the camera would be an entity, but it looks like it is always a `Transform`.

Comment (Kyjor) [reply src/Camera.jl:13]: Yep I definitely messed that comment up 😅

File: src/Component/CircleCollider.jl [hunk 2]
```diff
@@ -45,121 +46,122 @@ module CircleColliderModule
...
+                # if collision[1] == Right::CollisionDirection
+                #     push!(this.currentCollisions, colliders[i])
+                #     for eventToCall in this.collisionEvents
+                #         eventToCall()
+                #     end
+                #     #Begin to overlap, correct position
+                #     transform.setPosition(Math.Vector2f(transform.getPosition().x - collision[2], transform.getPosition().y))
+                # end
+                # if collision[1] == Bottom::CollisionDirection
+                #this.isGrounded = collision[1]
```
Comment (mrufsvold) [inline src/Component/CircleCollider.jl:117]: can this deadwood be removed?

File: src/Utils.jl [hunk 3]
```diff
@@ -10,4 +10,13 @@ function CallSDLFunction(func::Function, args...)
...
 
     return ret
-end
\ No newline at end of file
+end
+
+function deprecated_get_property(method_lookup, this::T, s::Symbol) where T
+    if haskey(method_lookup, s)
+        f = method_lookup[s]
+        # @warn "Using get_property to access the function $s from $T is deprecated. Please call $(typeof(f)) instead"
```
Comment (mrufsvold) [inline src/Utils.jl:18]: This is the log I referred to in my message. It BLOWS UP right now lol

File: src/Component/CircleCollider.jl (hunk 2 above)
Comment (Kyjor) [reply src/Component/CircleCollider.jl:117]: Yep! This is something I need to revisit, but it probably won't be for a while

**Comment by mrufsvold** [comment]
Pushed a doc string and kicked all the tests off again accidentally. Either way, ready for review!

File: src/Camera.jl (hunk 1 above)
Comment (mrufsvold) [reply src/Camera.jl:13]: Oh! Glad that was you not me! No worries!

**Comment by Kyjor** [comment]
This is awesome, thank you! Give me an hour or two and I can fully review it and run some manual tests that aren't covered (yet) by automation

**Review by Kyjor — APPROVED** [review]

**Pull request merged**